src/auth/test/axios.test.ts

```diff
@@ -0,0 +1,132 @@
+import {
+  accessToken,
+  BASE_URL,
+  invalidExp,
+  mockExpiredToken,
+  mockTokenResponse,
+  mockUserDataResponse,
+  validExp,
+} from '../../App.test';
+import { AdminApiClientRoutes } from '../../api/protectedApiClient';
+import { AxiosError } from 'axios';
+import authClient from '../authClient';
+import { responseErrorInterceptor } from '../axios';
+import store, { C4CState } from '../../store';
+import { AsyncRequestCompleted } from '../../utils/asyncRequest';
+import { generateState } from './thunks.test';
+import nock from 'nock';
+
+// a mock partial state with a valid refresh token
+const mockValidAuthState: Partial<C4CState> = {
+  authenticationState: {
+    tokens: AsyncRequestCompleted(mockTokenResponse(validExp)),
+    userData: AsyncRequestCompleted(mockUserDataResponse),
+  },
+};
+
+// a mock partial state with an invalid refresh token
+const mockInvalidAuthState: Partial<C4CState> = {
+  authenticationState: {
+    tokens: AsyncRequestCompleted(mockTokenResponse(invalidExp)),
+    userData: AsyncRequestCompleted(mockUserDataResponse),
+  },
+};
+
```

Comment:
📝 Fun fact - if you `jest.mock('../../store')` here and `npm run test`, tests in other modules will break. Presumably this is because all the modules depend on the mock, but running `jest.resetAllMocks` and `jest.unmock` before each test won't work...  My takeaway here is to mock as conservatively as possible.

Reply:
Yes! I personally hate the jest.mock(module) function.

We should ideally only use jest for its cool test functions and nothing else. Pass stateful dependencies as parameters!!